### PR TITLE
[TE] [notification] JIRA alerting framework

### PR DESCRIPTION
**Pull request by akshayrai**
Changes:
* Laid out the framework for filing alerts via JIRA tickets
* Added jira template for Metric Alerts - (TODO: template for Entity Alerts)
* Added unit test to validate the jira fields
* Moved some classes under the notification package

Configuration:
* Admin configuration - contains basic configs to connect to JIRA (user, passwd, host, def. jira project, issue type id)
* Client configuration - customize the jira fields in subscription config (assignee, labels, subject, project and issuetype)

Details:
* ticket is fired similar to the email alert - one consolidated ticket per subscription group
* `thirdeye` label is always appended to the ticket
* If no assignee is configured by user, then the ticket will remain `unassigned`
* Client configs like `project` and `issuetype` override the default admin configs

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/AlertContentFormatter.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package org.apache.pinot.thirdeye.notification.formatter.channels;
+
+import java.util.Properties;
+import org.apache.pinot.thirdeye.anomaly.ThirdEyeAnomalyConfiguration;
+import org.apache.pinot.thirdeye.datalayer.pojo.AlertConfigBean;
+import org.apache.pinot.thirdeye.notification.content.BaseNotificationContent;
+import org.apache.pinot.thirdeye.notification.formatter.ADContentFormatterContext;
+
+
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/AlertContentFormatter.java]: Maybe add some documentation for this class? What's the functionality for the class and how it should be used.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    issueBuilder.setFieldValue("labels", labels);
+
+    HtmlEmail email = new HtmlEmail();
+    String cid = "";
+    try {
+      if (StringUtils.isNotBlank(this.notificationContent.getSnaphotPath())) {
+        cid = email.embed(new File(this.notificationContent.getSnaphotPath()));
+      }
+    } catch (Exception e) {
+      LOG.error("Exception while embedding screenshot for anomaly", e);
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java]: Looks like this only plugs in the snapshot. When is the screenshot gets generated?

Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/JiraContentFormatter.java]: The screenshots are generated in the respective templates when you trigger notificationContent.format(...)

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/AlertContentFormatter.java (hunk 1 above)
Comment (akshayrai) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/notification/formatter/channels/AlertContentFormatter.java]: added

**Pull request merged**